api_calls.py

```diff
@@ -361,46 +365,40 @@ def upload_chapter(self, course, book, chapter, args):
             if not args.pdf.lower().endswith(".pdf"):
                 print("PDF argument must be a PDF file.")
                 return False
-            files.append(('input_file_pdf', open(self.expand_file_path(args.pdf), 'rb')))
+            files.append(('input_file_pdf', Path(args.pdf)))
             if args.video_url is not None:
                 update_params['video_url_pdf'] = args.video_url
         elif args.slides is not None:
             if not args.slides.lower().endswith(".pdf"):
                 print("Slides argument must be a PDF file.")
                 return False
-            files.append(('input_file_slide', open(self.expand_file_path(args.slides), 'rb')))
+            files.append(('input_file_slide', Path(args.slides)))
             if args.video_url is not None:
                 update_params['video_url_slide'] = args.video_url
         elif args.xml is not None:
             if not (args.xml.lower().endswith(".xml") or args.xml.lower().endswith(".mlx")):
                 print("XML argument must be an XML or MLX file.")
                 return False
-            files.append(('input_file_xml', open(self.expand_file_path(args.xml), 'rb')))
+            files.append(('input_file_xml', Path(args.xml)))
             if args.attach is not None:
-                tmpDir = tempfile.mkdtemp()
-                curPath = os.getcwd()
-                os.chdir(tmpDir)
                 for fg in args.attach:
-                    image_path = self.expand_file_path(fg)
-                    if os.path.isdir(image_path):
-                        for root, _, fi in os.walk(image_path):
-                            for name in fi:
-                                full_path = os.path.join(root, name)
-                                shutil.copyfile(full_path, os.path.basename(full_path))
-                    else:
-                        file_glob = glob.glob(image_path)
-                        if len(file_glob) == 0:
-                            print("Invalid input: ", fg)
-                            return False
-                        for f in file_glob:
-                            full_path = self.expand_file_path(f)
-                            shutil.copyfile(full_path, os.path.basename(full_path))
-                for f in os.listdir(os.getcwd()):
-                    print("Uploading file: {}".format(f))
-                    files.append(('image', open(f, 'rb')))
+                    for f in glob.glob(fg):
```

Comment:
This doesn't work if `fg` contains `~` -- maybe use the self.expand_file_path function here.